packages/network-connector/src/index.ts

```diff
@@ -1,47 +1,90 @@
 import { ConnectorUpdate } from '@web3-react/types'
 import { AbstractConnector } from '@web3-react/abstract-connector'
-import Web3ProviderEngine from 'web3-provider-engine'
-import CacheSubprovider from 'web3-provider-engine/subproviders/cache.js'
-import { RPCSubprovider } from '@0x/subproviders/lib/src/subproviders/rpc_subprovider' // https://github.com/0xProject/0x-monorepo/issues/1400
 import invariant from 'tiny-invariant'
 
 interface NetworkConnectorArguments {
   urls: { [chainId: number]: string }
   defaultChainId?: number
-  pollingInterval?: number
-  requestTimeoutMs?: number
+}
+
+// taken from ethers.js, compatible interface with web3 provider
+type AsyncSendable = {
+  isMetaMask?: boolean;
+  host?: string;
+  path?: string;
+  sendAsync?: (request: any, callback: (error: any, response: any) => void) => void;
+  send?: (request: any, callback: (error: any, response: any) => void) => void;
+}
+
+class MiniRpcProvider implements AsyncSendable {
+  public readonly isMetaMask: false = false
+  public readonly chainId: number
+  public readonly url: string
+  public readonly host: string
+  public readonly path: string
+
+  constructor(chainId: number, url: string) {
+    this.chainId = chainId
+    this.url = url
+    const parsed = new URL(url)
+    this.host = parsed.host
+    this.path = parsed.pathname
+  }
+
+  sendAsync(request: any, callback: (error: any, response: any) => void): void {
+    fetch(this.url, {
+      method: 'POST',
+      body: JSON.stringify(request)
+    })
+      .then(
+        result => {
+          return result.json()
```

Comment:
Missing `result.status` check here.  I believe calling `result.json()` on a not-OK response is likely to fail, and if it does succeed you won't get an object shaped like you expect.